src/Workspaces/Core/Portable/Workspace/Solution/TextLoader.cs

```diff
@@ -17,15 +17,23 @@ namespace Microsoft.CodeAnalysis
     /// </summary>
     public abstract class TextLoader
     {
+        private protected static readonly Workspace DummyWorkspace = new ImmutableWorkspace(Host.Mef.MefHostServices.DefaultHost, WorkspaceKind.Custom);
+        private static readonly DocumentId _dummyDocumentId = DocumentId.CreateNewId(ProjectId.CreateNewId("dummy"));
+
         private const double MaxDelaySecs = 1.0;
         private const int MaxRetries = 5;
         internal static readonly TimeSpan RetryDelay = TimeSpan.FromSeconds(MaxDelaySecs / MaxRetries);
 
         internal virtual string? FilePath => null;
 
+        internal Task<TextAndVersion> LoadTextAndVersionAsync(CancellationToken cancellationToken)
+            => LoadTextAndVersionAsync(DummyWorkspace, _dummyDocumentId, cancellationToken);
```

Comment:
doc... i don't quite get what this is doing.

Reply:
Just an overload we can call internally that calls the abstract method.